### PR TITLE
feat: Add live agent load report api

### LINKED ISSUE #4305: Live agent load report API
Create a reporting API to fetch the below metrics

1. Account level conversation metrics

| Open           | Unattended | Unassigned |
| ----------- | ------------ | ------------ |
|          10       |           5         |          5          |

2. User based conversation metrics for the account.

| User            | Open           | Unattended |
| ----------- | ----------- | ------------ |
|        Dev      |          10       |           5         |
|        Pranav      |          6       |           2         |

3. Filter conversation metrics by the user (Eg: Dev).

| User            | Open           | Unattended |
| ----------- | ----------- | ------------ |
|        Dev      |          10       |           5         |

@pranavrajs please share your thoughts on the API.

### PR DESCRIPTION
**Pull request by aswindevps**
## Description

This change allows the admin user to fetch conversation metrics for an account, agents, and filter conversation metrics for a specific agent.

Fixes #4305

## Type of change

- [ ] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Below scenarios verified using Postman.

1. Conversation metrics for account

Screenshot 2022-03-26 at 5 50 24 PM

2. Conversation metrics in user level

Screenshot 2022-03-26 at 5 52 26 PM

3. Conversation metrics for a specific user.

Screenshot 2022-03-26 at 5 53 55 PM

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/builders/v2/report_builder.rb [hunk 1]
```diff
@@ -121,4 +129,27 @@ def avg_first_response_time_summary
...
+      arr << {
+        user: { id: user.id, name: user.name, thumbnail: user.avatar_url },
+        metric: conversations
+      }
+    end
+  end
+
+  def conversations
+    @open_conversations = scope.conversations.open
+    first_response_count = scope.reporting_events.where(name: 'first_response', conversation_id: @open_conversations.pluck('id')).count
```
Comment (pranavrajs) [inline app/builders/v2/report_builder.rb:147]: This creates an IN query, I'm not sure whether it is performant. @sojan-official

**Review by pranavrajs — COMMENTED** [review]
Logic seems good, @sojan-official Please check the implementation here.

File: app/builders/v2/report_builder.rb (hunk 1 above)
Comment (sojan-official) [reply app/builders/v2/report_builder.rb:147]: discussed multiple options with @aswindevps  and amidst the ones which we came up to get this metric, this seemed the most efficient one.
- Even at 10000 open conversations for the agent, it still shouldn't be an immediate concern. 

ref: https://stackoverflow.com/a/47729784/939299

Comment (pranavrajs) [reply app/builders/v2/report_builder.rb:147]: Sounds good, can you please check the PR and see if it is good to merged?

**Review by sojan-official — APPROVED** [review]

**Comment by pranavrajs** [comment]
@aswindevps Some specs are failing, can you please take a look? Also, please add the specs to developer API documentation

**Pull request merged**